### DISCUSSION TITLE
New integration: netbox2monit

### DISCUSSION THREAD

**Discussion by thomas-mc-work** [netbox-community/netbox#15420] [General]
Hello NetBox community!

I've created an integration with [the monit monitoring solution](https://mmonit.com/monit/). It takes all physical and virtual machines and creates a configuration file for monit. This enables you to use NetBox as the single source of truth for your monitoring.

Here is the source repo: https://codeberg.org/thomas-mc-work/netbox2monit

Would it make sense to add that to your list of plugins? However actually this isn't a proper plugin. But probably still useful to a lot of users.

Thanks!

**Comment by MROSSFTEK** [comment]
think im using your module wrong...? i ran it and im getting this back.
your setup information is rather short, so perhaps im mising something?

Traceback (most recent call last):
  File "/netbox2monit/netbox2monit.py", line 74, in <module>
    execute_query(netbox_token, f'{netbox_base_url}/api/dcim/devices', ips_include, ips_exclude)
  File "/netbox2monit/netbox2monit.py", line 16, in execute_query
    if device_is_active(device) \
       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/netbox2monit/netbox2monit.py", line 31, in device_is_active
    return device['status']['value'] == 'active'
           ~~~~~~~~~~~~~~~~^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

**Comment by thomas-mc-work** [comment]
Thanks for the request! Could you please let know about the content of `response` within the function `execute_query`? You can either start it in debug mode and add a breakpoint or you execute the query manually (either using `curl` or any other client like Hoppscotch).

**Comment by MROSSFTEK** [comment]
forgive me.. not sure how to run it in debug mode.. im "typically" not a coder, so my skills sometimes lack a little.

**Comment by thomas-mc-work** [comment]
Ok, no worries! Could you insert these lines in line 14?:

    import pprint
    pprint.pprint(response)

… and then let know the output. Maybe via email, so we don't pollute this issue (thomas.mc.work@posteo.de).

**Comment by MROSSFTEK** [comment]
i added each of those lines at line 14 and 15, making the original line 14 down to line 16. 
ran the script and now i see

  File "/netbox2monit/netbox2monit.py", line 16
    (response) response = requests.request("GET", netbox_url, headers=headers).json()
               ^^^^^^^^
SyntaxError: invalid syntax

**Reply by thomas-mc-work** [reply]
Could you please execute the query on the CLI and post the response?

    curl -sH 'Authorization: Token my-token' "http://my-server.org/api/dcim/devices/"

- replace `my-token` with your API-token
- replace the hostname of the server
